tdrs-backend/tdpservice/data_files/admin.py

```diff
@@ -2,13 +2,47 @@
 from django.contrib import admin
 
 from ..core.utils import ReadOnlyAdminMixin
+from django.utils.translation import ugettext_lazy as _
 from .models import DataFile, LegacyFileTransfer
 from tdpservice.parsers.models import DataFileSummary, ParserError
 from django.conf import settings
 from django.utils.html import format_html
 
 DOMAIN = settings.FRONTEND_BASE_URL
 
+
+class NewestVersionFilter(admin.SimpleListFilter):
+    """Simple filter class to show newest created datafile records."""
+
+    title = _('Newest')
+
+    parameter_name = 'created_at'
+
+    def lookups(self, request, model_admin):
+        """Available options in dropdown."""
+        return (
+            (None, _('Most recent version')),
+            ('all', _('All')),
+        )
+
+    def choices(self, cl):
+        """Update query string based on selection."""
+        for lookup, title in self.lookup_choices:
+            yield {
+                'selected': self.value() == lookup,
+                'query_string': cl.get_query_string({
+                    self.parameter_name: lookup,
+                }, []),
+                'display': title,
+            }
+
+    def queryset(self, request, queryset):
+        """Sort queryset to show latest records."""
+        if self.value() is None and queryset.exists():
+            return queryset.order_by("stt__stt_code", "-pk")\
```

Comment:
Just a note: Postgres 'pk' is not always in order. Postgres caches pks and reserves them when assigning, and depending on which cached id is being assigned, pks might be assigned out of sync. However, in case TANF app it should be very rare, but I still suggest you use data/time instead.

Reply:
@elipe17 the ticket says "...adds that behavior to the Data Files page", which i assumed meant replicating the same behavior with regards to the stt filtering. as it is, the creation date filter also doesn't really sort on creation date or "most recent version" so i think it could do with a name change at the very least.

@victoriaatraft or @ADPennington could you elaborate a little on what the "most recent version" filter should do on the datafiles page?

Reply:
Hi @jtimpe. My apologies, I did not update the verbiage in the tasks. This should be on the DAC Index Page, not the data files page. 
Instead of filtering by Most Recent on the DAC Index Page, the wording of the label itself is changed to Most Recent. Hope that helps! ![Screenshot 2024-08-27 at 7 19 40 AM](https://github.com/user-attachments/assets/2c1caf6c-fa5e-4892-9327-012c5c2551d2)

Reply:
@jtimpe  i think we want the ability to filter the data files table by most recent version. (currently i believe the current filter is the version number?). And I believe we also want the default view of the table (when we navigate to this page) to show the most recent versions of the files. so that specific filter would already be set. but let us know if thats not feasible.